packages/eas-cli/src/commands/branch/publish.ts

```diff
@@ -332,32 +365,74 @@ export default class BranchPublish extends Command {
       }));
     }
 
-    let newUpdateGroup;
-    const publishSpinner = ora('Publishing...').start();
-    try {
-      newUpdateGroup = await PublishMutation.publishUpdateGroupAsync({
+    const runtimeToPlatformMapping: { [keyof: string]: string[] } = {};
+    for (const runtime of new Set(Object.values(runtimeVersions))) {
+      runtimeToPlatformMapping[runtime] = Object.entries(runtimeVersions)
+        .filter(pair => pair[1] === runtime)
+        .map(pair => pair[0]);
+    }
+
+    // Sort the updates into different groups based on their platform specific runtime versions
+    const updateGroups = Object.entries(runtimeToPlatformMapping).map(([runtime, platforms]) => {
+      const localUpdateInfoGroup = Object.fromEntries(
+        platforms.map(platform => [
+          platform,
+          unsortedUpdateInfoGroups[platform as keyof UpdateInfoGroup],
+        ])
+      );
+
+      if (republish && !oldRuntimeVersion) {
+        throw new Error(
+          'Can not find the runtime version of the update group that is being republished.'
+        );
+      }
+      return {
         branchId,
-        updateInfoGroup,
-        runtimeVersion: republish ? oldRuntimeVersion! : runtimeVersion,
+        updateInfoGroup: localUpdateInfoGroup,
+        runtimeVersion: republish ? oldRuntimeVersion : runtime,
         message,
-      });
+      };
+    });
+    let newUpdates;
+    const publishSpinner = ora('Publishing...').start();
+    try {
+      newUpdates = await PublishMutation.publishUpdateGroupAsync(updateGroups);
       publishSpinner.succeed('Published!');
     } catch (e) {
       publishSpinner.fail('Failed to published updates');
       throw e;
     }
 
     if (jsonFlag) {
-      Log.log(JSON.stringify(newUpdateGroup));
+      Log.log(JSON.stringify(newUpdates));
     } else {
-      Log.log(
-        formatFields([
-          { label: 'branch', value: branchName },
-          { label: 'runtime version', value: runtimeVersion },
-          { label: 'update group ID', value: newUpdateGroup.group },
-          { label: 'message', value: message },
-        ])
-      );
+      if (new Set(newUpdates.map(update => update.group)).size > 1) {
```

Comment:
I understand that you need Set to remove duplicates, but if just checking if it at least one you can just check newUpdates length

Reply:
We may have a *single* update group with multiple updates. In that case we don't want to print this message.

Thus checking `newUpdates.length` won't work.

Reply:
Ahh sorry, I read that incorrectly